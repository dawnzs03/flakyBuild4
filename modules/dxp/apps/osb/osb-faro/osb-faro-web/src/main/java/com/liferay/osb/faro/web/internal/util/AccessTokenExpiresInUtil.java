/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

/**
 * @author Ivica Cardic
 */
public class AccessTokenExpiresInUtil {

	public static long getExpiresIn() {
		return _expiresIn;
	}

	public static void removeExpiresIn() {
		_expiresIn = 0;
	}

	public static void setExpiresIn(long expiresIn) {
		_expiresIn = expiresIn;
	}

	private static long _expiresIn;

}