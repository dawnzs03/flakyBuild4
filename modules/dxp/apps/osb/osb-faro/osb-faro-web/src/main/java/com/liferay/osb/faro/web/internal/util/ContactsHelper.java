/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.constants.FieldMappingConstants;
import com.liferay.osb.faro.engine.client.model.Individual;
import com.liferay.osb.faro.engine.client.model.IndividualSegment;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.osb.faro.web.internal.constants.FaroConstants;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualDisplay;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualSegmentDisplay;
import com.liferay.osb.faro.web.internal.model.display.main.FaroEntityDisplay;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Shinn Lok
 */
@Component(service = ContactsHelper.class)
public class ContactsHelper {

	public FaroEntityDisplay getContactsEntityDisplay(
		FaroProject faroProject, String contactsEntityId, int type) {

		if (type == FaroConstants.TYPE_INDIVIDUAL) {
			Individual individual = _contactsEngineClient.getIndividual(
				faroProject, contactsEntityId, null);

			return new IndividualDisplay(individual);
		}

		IndividualSegment individualSegment =
			_contactsEngineClient.getIndividualSegment(
				faroProject, contactsEntityId, false);

		return new IndividualSegmentDisplay(individualSegment);
	}

	public String getOwnerType(int type) {
		if (type == FaroConstants.TYPE_ACCOUNT) {
			return FieldMappingConstants.OWNER_TYPE_ACCOUNT;
		}
		else if (type == FaroConstants.TYPE_INDIVIDUAL) {
			return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL;
		}

		return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL_SEGMENT;
	}

	@Reference
	private ContactsEngineClient _contactsEngineClient;

}