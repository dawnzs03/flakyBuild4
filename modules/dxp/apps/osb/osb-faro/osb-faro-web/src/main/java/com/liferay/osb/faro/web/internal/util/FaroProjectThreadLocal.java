/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.model.FaroProject;
import com.liferay.petra.lang.CentralizedThreadLocal;

/**
 * @author Marcellus Tavares
 */
public class FaroProjectThreadLocal {

	public static FaroProject getFaroProject() {
		return _faroProject.get();
	}

	public static void removeFaroProject() {
		_faroProject.remove();
	}

	public static void setFaroProject(FaroProject faroProject) {
		_faroProject.set(faroProject);
	}

	private static final ThreadLocal<FaroProject> _faroProject =
		new CentralizedThreadLocal<>(
			FaroProjectThreadLocal.class + "._faroProject");

}