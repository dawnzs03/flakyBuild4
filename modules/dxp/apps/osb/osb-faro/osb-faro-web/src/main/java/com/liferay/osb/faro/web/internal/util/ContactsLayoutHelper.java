/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.contacts.model.ContactsCardTemplate;
import com.liferay.osb.faro.contacts.service.ContactsCardTemplateLocalService;
import com.liferay.osb.faro.web.internal.card.template.ContactsCardTemplateManagerHelper;
import com.liferay.osb.faro.web.internal.model.display.contacts.card.template.ContactsLayoutTemplateSettingDisplay;
import com.liferay.petra.string.StringBundler;
import com.liferay.petra.string.StringPool;
import com.liferay.portal.kernel.model.UserConstants;

import java.util.ArrayList;
import java.util.List;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Matthew Kong
 */
@Component(service = ContactsLayoutHelper.class)
public class ContactsLayoutHelper {

	public String addContactsCardTemplateIds(long groupId, int type)
		throws Exception {

		List<List<ContactsLayoutTemplateSettingDisplay>>
			contactsLayoutTemplateSettingDisplaysList = new ArrayList<>();

		for (int[] contactsCardTemplateTypes :
				_contactsCardTemplateManagerUtil.
					getDefaultContactsCardTemplateTypes(type)) {

			List<ContactsLayoutTemplateSettingDisplay>
				contactsLayoutTemplateSettingDisplays = new ArrayList<>();

			for (int contactsCardTemplateType : contactsCardTemplateTypes) {
				ContactsCardTemplate contactsCardTemplate =
					_contactsCardTemplateLocalService.addContactsCardTemplate(
						groupId, UserConstants.USER_ID_DEFAULT,
						_contactsCardTemplateManagerUtil.getDefaultName(
							contactsCardTemplateType),
						_contactsCardTemplateManagerUtil.getDefaultSettings(
							contactsCardTemplateType),
						contactsCardTemplateType);

				contactsLayoutTemplateSettingDisplays.add(
					new ContactsLayoutTemplateSettingDisplay(
						contactsCardTemplate.getContactsCardTemplateId(), 0));
			}

			contactsLayoutTemplateSettingDisplaysList.add(
				contactsLayoutTemplateSettingDisplays);
		}

		return JSONUtil.writeValueAsString(
			contactsLayoutTemplateSettingDisplaysList);
	}

	public String addHeaderContactsCardTemplateIds(long groupId, int type) {
		int[] contactsCardTemplateTypes =
			_contactsCardTemplateManagerUtil.
				getDefaultHeaderContactsCardTemplateTypes(type);

		if (contactsCardTemplateTypes.length == 0) {
			return StringPool.BLANK;
		}

		StringBundler sb = new StringBundler(
			contactsCardTemplateTypes.length * 2);

		for (int contactsCardTemplateType : contactsCardTemplateTypes) {
			ContactsCardTemplate contactsCardTemplate =
				_contactsCardTemplateLocalService.addContactsCardTemplate(
					groupId, UserConstants.USER_ID_DEFAULT,
					_contactsCardTemplateManagerUtil.getDefaultName(
						contactsCardTemplateType),
					_contactsCardTemplateManagerUtil.getDefaultSettings(
						contactsCardTemplateType),
					contactsCardTemplateType);

			sb.append(contactsCardTemplate.getContactsCardTemplateId());

			sb.append(StringPool.COMMA);
		}

		sb.setIndex(sb.index() - 1);

		return sb.toString();
	}

	@Reference
	private ContactsCardTemplateLocalService _contactsCardTemplateLocalService;

	@Reference
	private ContactsCardTemplateManagerHelper _contactsCardTemplateManagerUtil;

}