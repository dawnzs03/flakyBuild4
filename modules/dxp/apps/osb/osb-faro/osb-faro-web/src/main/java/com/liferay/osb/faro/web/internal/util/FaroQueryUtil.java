/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.petra.string.StringPool;
import com.liferay.portal.kernel.util.StringUtil;
import com.liferay.portal.kernel.util.Validator;

import java.util.stream.Collectors;
import java.util.stream.IntStream;

/**
 * @author Gabriel Ibson
 */
public class FaroQueryUtil {

	public static String sanitizeQuery(String query) {
		if (Validator.isBlank(query)) {
			return query;
		}

		if (StringUtil.equalsIgnoreCase(query, StringPool.NULL)) {
			return "\\null";
		}

		IntStream intStream = query.codePoints();

		return intStream.mapToObj(
			c -> (char)c
		).map(
			c -> {
				if (_CHARACTERS_TO_BE_ESCAPED.indexOf(c) >= 0) {
					return "\\" + c;
				}

				return String.valueOf(c);
			}
		).collect(
			Collectors.joining()
		);
	}

	private static final String _CHARACTERS_TO_BE_ESCAPED =
		"+-=&&||><!(){}[]^\"~*?:\\/%";

}