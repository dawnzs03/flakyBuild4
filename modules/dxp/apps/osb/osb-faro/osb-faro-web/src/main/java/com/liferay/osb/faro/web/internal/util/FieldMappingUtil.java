/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.model.FieldMapping;
import com.liferay.osb.faro.engine.client.model.FieldMappingMap;
import com.liferay.osb.faro.engine.client.model.Results;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.petra.function.transform.TransformUtil;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * @author Matthew Kong
 */
public class FieldMappingUtil {

	public static List<FieldMappingMap> getNewFieldMappingMaps(
		ContactsEngineClient contactsEngineClient, FaroProject faroProject,
		String context, List<FieldMappingMap> fieldMappingMaps) {

		List<FieldMappingMap> newFieldMappingMaps = new ArrayList<>();

		Results<FieldMapping> results = contactsEngineClient.getFieldMappings(
			faroProject, context,
			TransformUtil.transform(fieldMappingMaps, FieldMappingMap::getName),
			1, 10000, null);

		Set<String> currentFieldNames = new HashSet<>();

		for (FieldMapping fieldMapping : results.getItems()) {
			currentFieldNames.add(fieldMapping.getFieldName());
		}

		Set<String> newFieldMappingNames = new HashSet<>();

		for (FieldMappingMap fieldMappingMap : fieldMappingMaps) {
			String name = fieldMappingMap.getName();

			if (currentFieldNames.contains(name) ||
				newFieldMappingNames.contains(name)) {

				continue;
			}

			newFieldMappingNames.add(name);

			newFieldMappingMaps.add(fieldMappingMap);
		}

		return newFieldMappingMaps;
	}

}