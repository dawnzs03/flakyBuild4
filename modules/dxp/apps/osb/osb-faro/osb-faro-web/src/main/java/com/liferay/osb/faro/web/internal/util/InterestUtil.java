/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.model.Interest;
import com.liferay.osb.faro.engine.client.model.Rels;
import com.liferay.osb.faro.engine.client.model.Results;
import com.liferay.osb.faro.engine.client.util.OrderByField;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.osb.faro.web.internal.model.display.FaroResultsDisplay;
import com.liferay.osb.faro.web.internal.model.display.contacts.InterestDisplay;

import java.util.List;
import java.util.function.Function;

/**
 * @author Shinn Lok
 */
public class InterestUtil {

	@SuppressWarnings("unchecked")
	public static FaroResultsDisplay getInterests(
		FaroProject faroProject, String channelId, String contactsEntityId,
		String query, int cur, int delta, List<OrderByField> orderByFields,
		ContactsEngineClient contactsEngineClient) {

		Results<Interest> results = contactsEngineClient.getInterests(
			faroProject, channelId, contactsEntityId, null, null, query,
			Rels.Interests.PAGES_VISITED, cur, delta, orderByFields);

		Function<Interest, InterestDisplay> function = InterestDisplay::new;

		return new FaroResultsDisplay(results, function);
	}

}