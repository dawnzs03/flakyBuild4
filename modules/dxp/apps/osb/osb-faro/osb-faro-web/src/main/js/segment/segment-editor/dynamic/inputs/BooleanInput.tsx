import autobind from 'autobind-decorator';
import Form from 'shared/components/form';
import React from 'react';
import {BOOLEAN_OPTIONS} from '../utils/constants';
import {Option, Picker} from '@clayui/core';
import {Property} from 'shared/util/records';

interface IBooleanInputProps {
	displayValue: string;
	id?: string;
	onChange: (object) => void;
	operatorRenderer: React.ElementType;
	property: Property;
	value: string;
}
export default class BooleanInput extends React.Component<IBooleanInputProps> {
	componentDidMount() {
		const {
			id,
			property: {entityName, type}
		} = this.props;

		if (!id) {
			analytics.track('Dynamic Segment Creation - Completed Attribute', {
				entityName,
				type
			});
		}
	}

	@autobind
	handleChange(value) {
		this.props.onChange({value});
	}

	render() {
		const {
			displayValue,
			operatorRenderer: OperatorDropdown,
			property: {entityName},
			value
		} = this.props;

		return (
			<div className='criteria-statement'>
				<Form.Group autoFit>
					<Form.GroupItem className='entity-name' label shrink>
						{entityName}
					</Form.GroupItem>

					<Form.GroupItem className='display-value' label shrink>
						{displayValue}
					</Form.GroupItem>

					<OperatorDropdown />

					<Form.GroupItem shrink>
						<Picker
							className='criterion-input'
							items={BOOLEAN_OPTIONS}
							onSelectionChange={this.handleChange}
							selectedKey={value}
						>
							{({label, value}) => (
								<Option key={value}>{label}</Option>
							)}
						</Picker>
					</Form.GroupItem>
				</Form.Group>
			</div>
		);
	}
}
