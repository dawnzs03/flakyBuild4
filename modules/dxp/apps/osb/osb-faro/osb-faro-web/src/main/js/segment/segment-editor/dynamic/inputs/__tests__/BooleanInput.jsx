import BooleanInput from '../BooleanInput';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {Property} from 'shared/util/records';

jest.unmock('react-dom');

describe('BooleanInput', () => {
	afterEach(cleanup);

	it('should render', () => {
		const {container, getByText} = render(
			<BooleanInput
				operatorRenderer={() => <div>{'operator'}</div>}
				property={new Property()}
			/>
		);
		fireEvent.click(getByText('Select an option'));

		expect(getByText('True')).toBeTruthy();
		expect(getByText('False')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('should render with data', () => {
		const {container, getAllByText, getByText} = render(
			<BooleanInput
				displayValue='Do Not Call'
				operatorRenderer={() => <div>{'operator'}</div>}
				property={new Property()}
				value='true'
			/>
		);
		fireEvent.click(getByText('True'));

		expect(getAllByText('True')[1]).toBeTruthy();
		expect(getByText('False')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});
});
