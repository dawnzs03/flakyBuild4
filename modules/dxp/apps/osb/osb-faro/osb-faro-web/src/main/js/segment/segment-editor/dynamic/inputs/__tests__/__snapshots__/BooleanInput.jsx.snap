// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanInput should render 1`] = `
<div>
  <div
    class="criteria-statement"
  >
    <div
      class="form-group form-group-autofit"
    >
      <div
        aria-hidden="true"
        class="form-group-item entity-name form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        aria-hidden="true"
        class="form-group-item display-value form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
      >
        operator
      </div>
      <div
        class="form-group-item form-group-item-shrink"
      >
        <button
          aria-activedescendant="true"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-2"
          class="form-control form-control-select form-control-select-secondary criterion-input show"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          Select an option
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`BooleanInput should render with data 1`] = `
<div>
  <div
    class="criteria-statement"
  >
    <div
      class="form-group form-group-autofit"
    >
      <div
        aria-hidden="true"
        class="form-group-item entity-name form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        aria-hidden="true"
        class="form-group-item display-value form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      >
        Do Not Call
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
      >
        operator
      </div>
      <div
        class="form-group-item form-group-item-shrink"
      >
        <button
          aria-activedescendant="true"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-4"
          class="form-control form-control-select form-control-select-secondary criterion-input show"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          True
        </button>
      </div>
    </div>
  </div>
</div>
`;
